storm-core/src/ui/public/js/visualization.js

```diff
@@ -16,434 +16,330 @@
-        for(var i = 0; i < v[":inputs"].length; i++) {
-            if(stream_checked(v[":inputs"][i][":sani-stream"]) 
-               && v[":inputs"][i][":component"] == component)
-                ret = true;
+
+        function parseEdge(edgeJson, sourceId) {
+            // Make this stream available
+            addAvailableStream(edgeJson[":stream"])
+
+            // Create a unique Id
+            var id = edgeJson[":component"] + ":" + sourceId + ":" + edgeJson[":stream"];
+
+            // Determine if stream is enabled
+            if (!isStreamEnabled(edgeJson[":stream"])) {
+                // Remove edge
+                edges.remove({ "id": id });
+                return
+            }
+
+            if (!edges.get(id)) {
+                console.log("Updating edge " + id );
+                edges.update({
+                    "id": id,
+                    "from": edgeJson[":component"],
+                    "to": sourceId,
+                    "label": edgeJson[":stream"],
+                    "title": "From: " + edgeJson[":component"] + "<br>To: " + sourceId + "<br>Grouping: " + edgeJson[":grouping"]
+                });
+            }
         }
-    });
-    return ret;
-}
-
-function gather_stream_count(stats, stream, time) {
-    var transferred = 0;
-    if(stats)
-        for(var i = 0; i < stats.length; i++) {
-            if(stats[i][":transferred"] != null && stats[i][":transferred"][time] != undefined)
-            {
-                var stream_trans = stats[i][":transferred"][time][stream];
-                if(stream_trans != null)
-                    transferred += stream_trans;
+
+        function addAvailableStream(streamId) {
+            // Create a master list of all streams
+            if (availableStreamsHash[streamId] == null) {
+                availableStreamsHash[streamId] = true;
+                updateAvailableStreams();
             }
         }
-    return transferred;
-}
-
-
-function rechoose(jdat, sys, box) {
-    var id = box.id;
-    if($(box).is(':checked'))
-    {
-        //Check each node in our json data to see if it has inputs from or outputs to selected streams. If it does, add a node for it.
-        $.each(jdat,function(k,v) {
-            if( has_checked_stream_input(v[":inputs"]) || has_checked_stream_output(jdat, k))
-                sys.addNode(k,v);
-        });
-           
-        //Check each node in our json data and add necessary edges based on selected components.
-        $.each(jdat, function(k, v) {
-            for(var i = 0; i < v[":inputs"].length; i++)
-                if(v[":inputs"][i][":sani-stream"] === id) {
-                    
-                    sys.addEdge(v[":inputs"][i][":component"], k, v);
+
+        function updateAvailableStreams() {
+            var container = jQuery("#available-streams");
+            for (var streamName in availableStreamsHash) {
+                var entry = jQuery(container).find("#stream-" + streamName)
```

Comment:
please test this out with a trident topology like org.apache.storm.starter.trident.TridentMinMaxOfDevicesTopology from the storm starter package.  The naming convention on the bolts/spouts is rather odd and messes with a lot of javascript/HTML.  Like for me I am seeing

```
Error: Syntax error, unrecognized expression: #stream-$batch  jquery-1.11.1.min.js:2:12720
	s</fb.error http://localhost:8080/js/jquery-1.11.1.min.js:2:12720
	s</fb.tokenize http://localhost:8080/js/jquery-1.11.1.min.js:2:18596
	fb http://localhost:8080/js/jquery-1.11.1.min.js:2:7068
	find http://localhost:8080/js/jquery-1.11.1.min.js:2:23593
	updateAvailableStreams http://localhost:8080/js/visualization.js:282:29
	addAvailableStream http://localhost:8080/js/visualization.js:275:17
	parseEdge http://localhost:8080/js/visualization.js:247:13
	parseNode http://localhost:8080/js/visualization.js:224:17
	parseResponse http://localhost:8080/js/visualization.js:101:17
	success http://localhost:8080/js/visualization.js:311:25
	j http://localhost:8080/js/jquery-1.11.1.min.js:2:27239
	fireWith http://localhost:8080/js/jquery-1.11.1.min.js:2:28057
	x http://localhost:8080/js/jquery-1.11.1.min.js:4:21841
	b http://localhost:8080/js/jquery-1.11.1.min.js:4:25897
```

I'm not sure what other places may be using stream/bolt names that need to be sanitized.